proxy/client_test.go

```diff
@@ -0,0 +1,213 @@
+package proxy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestExtractNextBatch(t *testing.T) {
+	input := `{
+		"next_batch": "s361093_69_4_8353_1",
+		"presence": {
+			"events": [
+				"event"
+			]
+		},
+		"rooms": {
+			"join": {
+				"!MbvBAvLuvGjNMUuRws:matrix.org": {}
+			}
+		}
+	}`
+
+	next_batch, err := extractNextBatch([]byte(input))
+	if err != nil {
+		t.Errorf("Expected no error, got '%v'", err)
+	}
+	if next_batch != "s361093_69_4_8353_1" {
+		t.Errorf("Expected next batch 's361093_69_4_8353_1', got '%v'",
```

Comment:
%q is your friend, maybe worth %qing both the want and got, and passing the want-value as a format arg, for consistency.